src/app/containers/SignupContainer/useSignup.js

```diff
@@ -211,10 +213,10 @@ const useSignup = (onLogin, { coupon, invite, availablePlans = VPN_PLANS } = {},
     const getToken = ({ coupon, invite, verificationToken, paymentDetails }) => {
         if (invite) {
             return { Token: `${invite.selector}:${invite.token}`, TokenType: TOKEN_TYPES.INVITE };
-        } else if (coupon) {
-            return { Token: coupon.code, TokenType: TOKEN_TYPES.COUPON };
         } else if (paymentDetails) {
             return { Token: paymentDetails.VerifyCode, TokenType: TOKEN_TYPES.PAYMENT };
+        } else if (coupon) {
```

Comment:
Coupon used to take priority, and payment details would not be applied, but there would still be some amount to pay, so it failed to signup. This is an oversight because I was testing with coupons that had 100% discount...